modules/tinymce/src/core/main/ts/keyboard/NavigationUtils.ts

```diff
@@ -72,13 +78,21 @@ const moveVertically = (editor: Editor, direction: LineWalker.VDirection, range:
   const caretPosition = CaretUtils.getNormalizedRangeEndPoint(direction, editor.getBody(), range);
   const caretClientRect = ArrUtils.last(caretPosition.getClientRects());
   const forwards = direction === LineWalker.VDirection.Down;
+  const root = editor.getBody();
 
   if (!caretClientRect) {
     return Optional.none();
   }
 
+  if (isCefAtEdgeSelected(editor)) {
+    const caretPosition = forwards ? CaretPosition.fromRangeEnd(range) : CaretPosition.fromRangeStart(range);
+    return (!forwards ? LineReader.getClosestPositionAbove(root, caretPosition) : LineReader.getClosestPositionBelow(root, caretPosition))
```

Comment:
This line has a fair bit of complexity with the ternary as well. I'd suggest we split this up over 2 lines which will also help with readability. For example:
```suggestion
    const getClosestFn = !forwards ? LineReader.getClosestPositionAbove : LineReader.getClosestPositionBelow;
    return getClosestFn(root, caretPosition)
```